src/Mod/Surface/App/Blending/BlendCurve.cpp

```diff
@@ -0,0 +1,170 @@
+/***************************************************************************
+ *   Copyright (c) 2022 Matteo Grellier <matteogrellier@gmail.com>         *
+ *                                                                         *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "../PreCompiled.h"
+#ifndef _PreComp_
+#include <BRepBuilderAPI_MakeEdge.hxx>
+#include <BSplCLib.hxx>
+#include <Base/Persistence.h>
+#include <Base/Vector3D.h>
+#include <GeomAPI_PointsToBSplineSurface.hxx>
+#include <Geom_BSplineSurface.hxx>
+#include <Geom_BezierCurve.hxx>
+#include <Mod/Part/App/Geometry.h>
+#include <Precision.hxx>
+#include <Standard_Version.hxx>
+#include <TColStd_Array1OfReal.hxx>
+#include <TColgp_Array1OfPnt.hxx>
+#include <TopoDS.hxx>
+#include <gp_Pnt.hxx>
+#include <math_Gauss.hxx>
+#include <math_Matrix.hxx>
+#endif
+#include "BlendCurve.h"
+
+using namespace Surface;
+
+BlendCurve::BlendCurve()
+{
+}
+
+BlendCurve::BlendCurve(std::vector<BlendPoint> blendPointsList)
+{
+    // Retrieve number of blendPoints and push them into blendPoints.
+    int nb_pts = blendPointsList.size();

```

Comment:
Either use size_t or auto but not int as otherwise this will cause compiler warnings on several platforms.

Reply:
Fixed.